src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java

```diff
@@ -0,0 +1,238 @@
+
+    // Capture log output to make sense of most failures
+    Logger cloudLogger =
+        LogManager.getLogManager()
+            .getLogger("com.google.jenkins.plugins.computeengine.ComputeEngineCloud");
+    if (cloudLogger != null) {
+      cloudLogger.addHandler(sh);
+    }
+
+    assertEquals(0, r.jenkins.clouds.size());
+    r.jenkins.clouds.add(gcp);
+    assertEquals(1, r.jenkins.clouds.size());
+
+    // Get a compute client for out-of-band calls to GCE
+    ClientFactory clientFactory = new ClientFactory(r.jenkins, new ArrayList<>(), projectId);
+    ComputeClient client = clientFactory.compute();
+    assertNotNull("ComputeClient can not be null", client);
+
+    // Other logging
+    Logger clientLogger =
+        LogManager.getLogManager()
+            .getLogger("com.google.jenkins.plugins.computeengine.ComputeClient");
+    if (clientLogger != null) {
+      clientLogger.addHandler(sh);
+    }
+
+    deleteIntegrationInstances(true, client, label, log);
+    return client;
+  }
+
+  static void teardown(
+      StreamHandler sh,
+      ByteArrayOutputStream logOutput,
+      ComputeClient client,
+      Map<String, String> label,
+      Logger log)
+      throws IOException {
+    log.info("teardown");
+    deleteIntegrationInstances(false, client, label, log);
+    sh.close();
+    log.info(logOutput.toString());
+  }
+
+  static InstanceConfiguration instanceConfiguration(
+      String startupScript,
+      String numExecutors,
+      String jenkinsLabels,
+      Map<String, String> label,
+      boolean createSnapshot,
+      boolean oneShot,
+      String template) {
+    InstanceConfiguration ic =
+        new InstanceConfiguration(
+            NAME_PREFIX,
+            REGION,
+            ZONE,
+            MACHINE_TYPE,
+            numExecutors,
+            startupScript,
+            PREEMPTIBLE,
+            MIN_CPU_PLATFORM,
+            jenkinsLabels,
+            CONFIG_DESC,
+            BOOT_DISK_TYPE,
+            BOOT_DISK_AUTODELETE,
+            BOOT_DISK_IMAGE_NAME,
+            BOOT_DISK_PROJECT_ID,
+            BOOT_DISK_SIZE_GB_STR,
+            false,
+            "",
+            "",
+            createSnapshot,
+            null,
+            new AutofilledNetworkConfiguration(NETWORK_NAME, SUBNETWORK_NAME),
+            EXTERNAL_ADDR,
+            false,
+            NETWORK_TAGS,
+            SERVICE_ACCOUNT_EMAIL,
+            RETENTION_TIME_MINUTES_STR,
+            LAUNCH_TIMEOUT_SECONDS_STR,
+            NODE_MODE,
+            new AcceleratorConfiguration(ACCELERATOR_NAME, ACCELERATOR_COUNT),
+            RUN_AS_USER,
+            oneShot,
+            template);
+    ic.appendLabels(label);
+    return ic;
+  }
+
+  static Map<String, String> getLabel(Class testClass) {
+    // GCE labels can only be lower case letters, numbers, or dashes
+    return ImmutableMap.of(testClass.getSimpleName().toLowerCase(), "delete");
```

Comment:
Could you please add a note clarifying the hard coded value "delete".

Reply:
Done